editor/src/core/shared/math-utils.ts

```diff
@@ -75,27 +75,48 @@ export function size(width: number, height: number): Size {
   }
 }
 
-export type RectangleInner = {
+export type SimpleRectangle = {
   x: number
   y: number
   width: number
   height: number
 }
 
-export type SimpleRectangle = RectangleInner
-export type Rectangle<C extends CoordinateMarker> = RectangleInner & C
-export type WindowRectangle = WindowModifier & RectangleInner
-export type CanvasRectangle = CanvasModifier & RectangleInner
-export type LocalRectangle = LocalModifier & RectangleInner
-export type NodeGraphRectangle = NodeGraphModifier & RectangleInner
+export type Rectangle<C extends CoordinateMarker> = SimpleRectangle & C
+export type WindowRectangle = WindowModifier & SimpleRectangle
+export type CanvasRectangle = CanvasModifier & SimpleRectangle
+export type LocalRectangle = LocalModifier & SimpleRectangle
+export type NodeGraphRectangle = NodeGraphModifier & SimpleRectangle
+
+export type InfinityRectangle<C extends CoordinateMarker> = { type: 'INFINITY_RECTANGLE' } & C
+
+const infinityRectangle = { type: 'INFINITY_RECTANGLE' }
+export const infinityCanvasRectangle = infinityRectangle as InfinityRectangle<CanvasModifier>
+export const infinityLocalRectangle = infinityRectangle as InfinityRectangle<LocalModifier>
+
+export type MaybeInfinityRectangle<C extends CoordinateMarker> = Rectangle<C> | InfinityRectangle<C>
+export type MaybeInfinityCanvasRectangle = MaybeInfinityRectangle<CanvasModifier>
+export type MaybeInfinityLocalRectangle = MaybeInfinityRectangle<LocalModifier>
+
+export function isInfinityRectangle<C extends CoordinateMarker>(
+  value: MaybeInfinityRectangle<C>,
+): value is InfinityRectangle<C> {
+  return 'type' in value && value.type === 'INFINITY_RECTANGLE'
+}
+
+export function isNonInfinityRectangle<C extends CoordinateMarker>(
```

Comment:
```suggestion
export function isFiniteRectangle<C extends CoordinateMarker>(
```

Reply:
Well yeah, sure, I guess in hindsight that probably makes way more sense 😁